yarn-project/foundation/src/fields/fields.test.ts

```diff
@@ -31,6 +31,14 @@ describe('GrumpkinScalar Serialization', () => {
 
     // Check if the deserialized instance is equal to the original
     expect(deserialized).toEqual(original);
+
+    // Note odd number of digits
+    const arbitraryString = '123';
+    const arbitraryHexString = '0x123';
+    const expectedBigInt = 291n;
+
+    expect(GrumpkinScalar.fromString(arbitraryString).toBigInt()).toEqual(expectedBigInt);
```

Comment:
shouldn't arbitary string resolve to 123 and not 291?

Reply:
hmm can we make it clear then that `fromString` expects a hex string?

Reply:
@rahul-kothari @sklppy88 treating `123` as `0x123` in `Fr.fromString`(and others) is a footgun. And it differs from the behaviour of native `BigInt`, `ethers`, `viem`, you name it.

Reply:
@sklppy88 why not mimic the behavior of the native `bigint`:
- if string starts with 0x, parse as hex
- else parse as decimal
?

It is the most logical solution to me.

If this is not an option, I would rather throw an error for non-prefixed hex strings.

Reply:
@sklppy88 can this problem get revised? I am hitting this issue in a deep object, so i cant easily convert all my strings to hex format

Reply:
@sklppy88  yeah, that would definitely help

Reply:
@sklppy88 i am have to do something like this:
```ts
encodeArguments(abi, args.map((arg) => {
  if (Number.isFinite(Number(arg))) {
    return new Fr(BigInt(arg));
  }
  return arg;
}));
```

Now imagine that args contains a nested object or array. i will have to iterate over each object deeply (prob using recursion) to convert each number into a hex string manually. 

Your example won't help as it will just change:
```diff
- return new Fr(BigInt(arg));
+ return new Fr(arg, "dec");
```

But if you fixed the original problem, i would be able to just write this code:
```ts
encodeArguments(abi)
```

Reply:
could you please explain the reasoning behind not wanting to change it to autodetection of decimal vs hex? For me it's such an obvious decision. What am i missing?